packages/round-manager/src/features/api/services/grantApplication.ts

```diff
@@ -82,61 +158,57 @@ export const grantApplicationApi = api.injectEndpoints({
     }),
     updateGrantApplication: builder.mutation<
       string,
-      {
-        id: string,
-        roundId: string,
-        status: "APPROVED" | "REJECTED",
-        projectsMetaPtr: MetadataPointer,
-        payoutAddress: string,
-        signerOrProvider: any
-      }
+      { roundId: string, application: GrantApplication, signer: any, provider: any }
     >({
-      queryFn: async ({ id, roundId, status, projectsMetaPtr, payoutAddress, signerOrProvider }) => {
+      queryFn: async ({ roundId, application, signer, provider }) => {
         try {
-          let reviewedApplications: any = []
-          let foundEntry = false
-
-          // read data from ipfs
-          if (projectsMetaPtr) {
-            reviewedApplications = await fetchFromIPFS(projectsMetaPtr.pointer)
-
-            // if grant application is already reviewed overwrite the entry
-            foundEntry = reviewedApplications.find((o: any, i: any) => {
-              if (o.id === id) {
-                reviewedApplications[i] = { id, status, payoutAddress }
-                return true // stop searching
-              }
-              return false
-            })
-          }
-
-          // create a new reviewed application entry
-          if (!foundEntry || !projectsMetaPtr) {
-            reviewedApplications.push({ id, status, payoutAddress })
-          }
-
-          // pin new list IPFS
-          const resp = await pinToIPFS({
-            content: reviewedApplications,
-            metadata: {
-              name: "reviewed-applications"
-            }
-          })
-          console.log("✅  Saved data to IPFS:", resp.IpfsHash)
+          const ipfsHash = await updateApplicationList([application], roundId, provider)
 
           // update projects meta pointer in round implementation contract
-          projectsMetaPtr = {
+          application.projectsMetaPtr = {
```

Comment:
nitpick: does this need to modify `application.projectsMetaPtr`, instead of using a new variable or just inlining directly in L179?

Reply:
i will inline this